components/project-card/meta/creators.jsx

```diff
@@ -7,18 +7,39 @@ const Creators = (props) => {
 
   let classnames = classNames(`creator d-block open-sans`, props.className);
   let labelText = props.showLabelText ? `Created by ` : ``;
+  let joinedCreators;
 
-  return <p className="my-2"><small className={classnames}>{labelText}{props.creators.join(`, `)}</small></p>;
+  if (typeof props.creators[0] === `string` ) {
+    joinedCreators = props.creators.join(`, `);
+  } else {
+    let creators = [];
+    for (let creator of props.creators){
+      // So that creators which are without a profile aren't links
+      let url = null;
+      if(typeof creator.profile_id === `number` && props.makeLink) {
+        url = `/profile/${creator.profile_id}`;
+      }
+      creators.push(<a key={creators.length} href={url}>{creator.name}</a>);
+    }
+    // Because react doesn't like to render comma separated JSX objects easily
+    joinedCreators = creators.slice(1).reduce((prev, current)=>{
+      return prev.concat([`, `, current]);
+    }, [creators[0]]);
+
+  }
+  return <p className="my-2"><small className={classnames}>{labelText}{joinedCreators}</small></p>;
```

Comment:
This might need a rewrite to simply do a map off of `props.creators` so that it just does "the right thing" based on the type, along the lines of
```
return props.creators.maps(data => {
  if (typeof data === `string`) {
    return <span className="creator" key={data}>{data}</span>;
  }
  let url = null;
  if(typeof data .profile_id === `number` && props.makeLink) {
    url = `/profile/${data.profile_id}`;
  }
  return <a className="creator" key={data.profile_id} href={url}>{data.name}</a>);
});
```
And then have CSS automatically inject that comma using a `creator + creator:before { content: ", "; }` rule.

Reply:
CSS didn't feel like a good idea for separation of concerns reasons. I ended up using `reduce` to squash them together and that works well enough.